tracer/src/Datadog.Trace/Debugger/LineProbeResolver.cs

```diff
@@ -105,31 +106,41 @@ private Trie CreateTrieForSourceFilePaths(Assembly loadedAssembly)
             return trie;
         }
 
-        private Assembly FindAssemblyContainingFile(string sourceFileFullPath)
+        private bool TryFindAssemblyContainingFile(string sourceFileFullPath, out string sourceFileFullPathFromPdb, out Assembly assembly)
         {
             lock (_locker)
             {
-                foreach (var assembly in AppDomain.CurrentDomain.GetAssemblies())
+                foreach (var candidateAssembly in AppDomain.CurrentDomain.GetAssemblies())
                 {
-                    var trie = GetSourceFilePathForAssembly(assembly);
+                    var trie = GetSourceFilePathForAssembly(candidateAssembly);
+                    if (trie == null)
+                    {
+                        continue;
+                    }
 
-                    if (trie != null && trie.ContainsPrefix(GetReversePath(sourceFileFullPath)))
+                    // Because we're actually interested in matching the ending of the string rather than the beginning,
+                    // we reverse the string before inserting or querying the trie, and then reverse it again when we get a string out.
+                    var reversedPath = trie.GetStringStartingWith(GetReversePath(sourceFileFullPath));
+                    if (reversedPath != null)
                     {
-                        return assembly;
+                        sourceFileFullPathFromPdb = GetReversePath(reversedPath);
+                        assembly = candidateAssembly;
+                        return true;
                     }
                 }
             }
 
-            return null;
+            sourceFileFullPathFromPdb = null;
+            assembly = null;
+            return false;
         }
 
         public LineProbeResolveResult TryResolveLineProbe(ProbeDefinition probe, out BoundLineProbeLocation location)
         {
             location = null;
-            var assembly = FindAssemblyContainingFile(probe.Where.SourceFile);
-            if (assembly == null)
+            if (!TryFindAssemblyContainingFile(probe.Where.SourceFile, out var filePathFromPdb, out var assembly))
             {
-                var message = $"Could not find a source file location for probe {probe.Id}.";
+                var message = $"Source file location for probe {probe.Id} was not found, possibly because the relevant assembly was not yet loaded.";
                 Log.Information(message);
```

Comment:
Not directly related to this PR, but you shouldn't use string interpolation with Serilog. Serilog takes a hash of the message template and uses that to cache string replacements; using interpolation explodes the cache, causing more work for all logging statements.

I've just updated this one, but _all_ logging statements should be "fixed" like this unfortunately 🙁 
```suggestion
                Log.Information("Source file location for probe {ProbeId} was not found, possibly because the relevant assembly was not yet loaded.", probe.Id);
```

Reply:
TIL, thank you! 

> I've just updated this one, but all logging statements should be "fixed" like this unfortunately 🙁

Perhaps we should consider adding https://github.com/Suchiman/SerilogAnalyzer and making [Serilog004](https://github.com/Suchiman/SerilogAnalyzer) an error-level static check?

Reply:
> Perhaps we should consider adding https://github.com/Suchiman/SerilogAnalyzer

Excellent idea, though we'll need to vendor it and update as the `Log` here isn't the `Serilog.Log` it's looking for 😄 

(That's part of the reason we have the issue in the first place - Rider, for example, has a similar analyzer built-in)

Reply:
Fixed remaining issues in 6583a343d8bfb23d29bd8bd5067c6cd8f2541391. Will create a JIRA ticket for the idea of adding a static check.